File: runtime/browser.go
package runtime

import "github.com/pkg/browser"

// Browser exposes browser methods to the runtime
type Browser struct{}

// NewBrowser creates a new runtime Browser struct
func NewBrowser() *Browser {
	return &Browser{}
}

// OpenURL opens the given url in the system's default browser
func (r *Browser) OpenURL(url string) error {
	return browser.OpenURL(url)
}

// OpenFile opens the given file in the system's default browser
func (r *Browser) OpenFile(filePath string) error {
	return browser.OpenFile(filePath)
}

File: internal/runtime/js/runtime/browser.js
/*
 _       __      _ __    
| |     / /___ _(_) /____
| | /| / / __ `/ / / ___/
| |/ |/ / /_/ / / (__  ) 
|__/|__/\__,_/_/_/____/  
The lightweight framework for web-like apps
(c) Lea Anthony 2019-present
*/
/* jshint esversion: 6 */

/**
 * Opens the given URL in the system browser
 *
 * @export
 * @param {string} url
 * @returns
 */
function OpenURL(url) {
	return window.wails.Browser.OpenURL(url);
}

/**
 * Opens the given filename using the system's default file handler
 *
 * @export
 * @param {sting} filename
 * @returns
 */
function OpenFile(filename) {
	return window.wails.Browser.OpenFile(filename);
}

module.exports = {
	OpenURL: OpenURL,
	OpenFile: OpenFile
};